### ISSUE TITLE
[MDEP-257] <includeClassifiers /> seems to have no effect

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-dependency-plugin#1102]
**[Patrick Aikens](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=duckpuppy)** opened **[MDEP-257](https://issues.apache.org/jira/browse/MDEP-257?redirect=false)** and commented

I'm trying to package some installer-related files from a project for use in a different izpack installer project.  I want to unpack them into the target/staging directory for inclusion in the izpack installer.  All of the dependencies I want to unpack have a classifier of "izpack".  Using the following configuration, the dependency plugin unpacks ALL dependencies, not just the ones with an "izpack" classifier.

```xml
<plugin>
  <artifactId>maven-dependency-plugin</artifactId>
  <version>2.1</version>
  <executions>
    <execution>
      <id>izpack</id>
      <phase>generate-resources</phase>
      <goals>
        <goal>unpack-dependencies</goal>
      </goals>
      <configuration>
        <outputDirectory>${project.build.directory}/staging</outputDirectory>
        <includeClassifiers>izpack</includeClassifiers>
      </configuration>
    </execution>
  </executions>
</plugin>
```

---

**Affects:** 2.1

**Comment by jira-importer** [comment]
**[Richard Vítek](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=nara)** commented

I encoutered the same problem, so I took a closer look at it and the problem lies in one of dependencies of this plugin, specifically maven-common-artifact-filters.
Update the version to current version 1.2 and replace usages of class TransitivityFilter for ProjectTransitivityFilter, and you'll be fine.

In detail, problem lies in class AbstractArtifactFeatureFilter, where in version 1.0 of the maven-common-artifact-filters plugin, NPE was thrown, caught then by FilterArtifact class, so the ClassifierFilter had no effect.
